source/kubernetes/create-app/index.html.md

```diff
@@ -0,0 +1,191 @@
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_GOVUK_ECR_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_GOVUK_ECR_SECRET_ACCESS_KEY }}
+  trigger-deploy:
+    name: Trigger deploy to ${{ inputs.environment || 'integration' }}
+    needs: build-and-publish-image
+    uses: alphagov/govuk-infrastructure/.github/workflows/deploy.yml@main
+    with:
+      imageTag: ${{ needs.build-and-publish-image.outputs.imageTag }}
+      environment: ${{ inputs.environment || 'integration' }}
+    secrets:
+      WEBHOOK_TOKEN: ${{ secrets.GOVUK_ARGO_EVENTS_WEBHOOK_TOKEN }}
+      WEBHOOK_URL: ${{ secrets.GOVUK_ARGO_EVENTS_WEBHOOK_URL }}
+      GH_TOKEN: ${{ secrets.GOVUK_CI_GITHUB_API_TOKEN }}
+```
+
+## Add image tag files to `govuk-helm-charts`
+
+[GOV.UK Helm Charts](https://github.com/alphagov/govuk-helm-charts) uses image tag files to keep
+track of the image used for the active deployment in each environment. Add an initial tag file named
+after your application into the `image-tags` folder for each of the four environments, i.e.
+`charts/app-config/image-tags/{test,integration,staging/production}/example`:
+
+```
+image_tag: v1
+automatic_deploys_enabled: true
+promote_deployment: true
+```
+
+## Add initial secrets to AWS Secrets Manager
+
+### Sentry
+
+In order to set up Sentry for your app, you will need a _DSN_ from the Sentry UI. This must be
+manually added as a key/value pair to the `govuk/common/sentry` secret in all three environments.
+
+You can use `gds-cli aws govuk-{integration/staging/production}-poweruser -l` to log in to the AWS
+Console for each environment and make the change.
+
+## Add app to `app-config` chart values
+
+### Start with integration
+
+Start off by adding your app to the integration environment by adding it to the array of application
+Helm chart values in `charts/app-config/values-integration.yaml`. This can be as simple as:
+
+```yml
+- name: example
+```
+
+This will use the [generic-govuk-app][generic-app] Helm chart to set up your application. You can
+find the default values in `values.yaml` in the chart folder, and amend your configuration's
+`helmValues` value to override them.
+
+For example, if your app needs a `RETICULATE_SPLINES` environment variable set to "1" in a given
+environment, your integration configuration could look like this:
+
+```yml
+- name: example
+  helmValues:
+    extraEnv:
+      - name: RETICULATE_SPLINES
+        value: 1
+```
+
+There are configuration options available for a wide range of different needs your application may
+have, from mounting volumes to disabling the default asset upload.
+
+### Validate your deployment
+
+You can now visit [Integration Argo CD][argo] and after a short wait, your new application will be
+available to find on the dashboard.
+
+When your application has successfully deployed, you can try to access it through `kubectl`'s
+[port forwarding feature](/kubernetes/fix-app/#port-forwarding-to-a-pod).
```

Comment:
Ah yeah, it's difficult to know where to stop here. I'd say this might be confusing, as most apps should have an ingress associated which would just make them accessible.